library/core/src/ffi/c_str.rs

```diff
@@ -296,6 +299,31 @@ impl CStr {
     #[must_use]
     #[stable(feature = "rust1", since = "1.0.0")]
     #[rustc_const_stable(feature = "const_cstr_from_ptr", since = "1.81.0")]
+    // Preconditions:
+    // - ptr must be non-null and properly aligned
+    // - ptr must point to a valid nul-terminated string within isize::MAX bytes
+    // - memory range must be valid for reads
+    #[requires(!ptr.is_null() && {
+        let mut next = ptr;
+        let mut found_null = false;
+        while can_dereference(next) {
+            if unsafe { *next == 0 } {
+                found_null = true;
+                break;
+            }
+            next = next.wrapping_add(1);
+        }
+        found_null
```

Comment:
do you mind moving this to a private aux function that you can reuse in other contracts, like the one from https://github.com/model-checking/verify-rust-std/pull/193?